doc/source/user/how-to-io.rst

```diff
@@ -0,0 +1,321 @@
+Write to a file to be read back by NumPy
+******************************************************************************
+
+Binary
+===============================================================================
+
+Use
+:func:`numpy.save`, or to store multiple arrays :func:`numpy.savez`
+or :func:`numpy.savez_compressed`.
+
+For :ref:`security and portability <how-to-io-pickle-file>`, set
+``allow_pickle=False`` unless the dtype contains Python objects, which
+requires pickling.
+
+Masked arrays :any:`can't currently be saved <MaskedArray.tofile>`,
+nor can other arbitrary array subclasses.
+
+Human-readable
+==============================================================================
+
+:func:`numpy.save` and :func:`numpy.savez` create binary files. To **write a
+human-readable file**, use :func:`numpy.savetxt`. The array can only be 1- or
+2-dimensional, and there's no ` savetxtz` for multiple files.
+
+Large arrays
+==============================================================================
+
+See :ref:`how-to-io-large-arrays`.
+
+******************************************************************************
+Read an arbitrarily formatted binary file ("binary blob")
+******************************************************************************
+
+Use a :doc:`structured array <basics.rec>`.
+
+**Example:**
+
+The ``.wav`` file header is a 44-byte block preceding ``data_size`` bytes of the
+actual sound data::
+
+    chunk_id         "RIFF"
+    chunk_size       4-byte unsigned little-endian integer
+    format           "WAVE"
+    fmt_id           "fmt "
+    fmt_size         4-byte unsigned little-endian integer
+    audio_fmt        2-byte unsigned little-endian integer
+    num_channels     2-byte unsigned little-endian integer
+    sample_rate      4-byte unsigned little-endian integer
+    byte_rate        4-byte unsigned little-endian integer
+    block_align      2-byte unsigned little-endian integer
+    bits_per_sample  2-byte unsigned little-endian integer
+    data_id          "data"
+    data_size        4-byte unsigned little-endian integer
+
+The ``.wav`` file header as a NumPy structured dtype::
+
+    wav_header_dtype = np.dtype([
+        ("chunk_id", (bytes, 4)), # flexible-sized scalar type, item size 4
+        ("chunk_size", "<u4"),    # little-endian unsigned 32-bit integer
+        ("format", "S4"),         # 4-byte string
+        ("fmt_id", "S4"),
+        ("fmt_size", "<u4"),
+        ("audio_fmt", "<u2"),     #
+        ("num_channels", "<u2"),  # .. more of the same ...
+        ("sample_rate", "<u4"),   #
+        ("byte_rate", "<u4"),
+        ("block_align", "<u2"),
+        ("bits_per_sample", "<u2"),
+        ("data_id", "S4"),
+        ("data_size", "u4"),
+        #
+        # the sound data itself cannot be represented here:
+        # it does not have a fixed size
+       ])
+
+    data = np.fromfile(f,dtype=wave_header_dtype)

```

Comment:
nit:
```suggestion
    ])

    data = np.fromfile(f, dtype=wave_header_dtype)

```

Also, this is a bad example - if you're reading only a header, you should probably use `np.fromfile(f, dtype=wave_header_dtype, count=1)[0]`